src/System.Runtime.Serialization.Xml/tests/SerializationTypes.cs

```diff
@@ -2908,3 +2908,34 @@ public class TypeOfReferenceChild
     [DataMember]
     public string Name { get; set; }
 }
+
+[DataContract]
+public sealed class TypeWithInternalDefaultConstructor
+{
+    internal TypeWithInternalDefaultConstructor()
+    {
+    }
+
+    internal static TypeWithInternalDefaultConstructor CreateInstance()
+    {
+        return new TypeWithInternalDefaultConstructor();
+    }
+
+    [DataMember]
+    public string Name { get; set; }
+}
+
+public sealed class TypeWithInternalDefaultConstructorWithoutDataContractAttirbute
```

Comment:
>TypeWithInternalDefaultConstructorWithoutDataContractAttirbute [](start = 20, length = 62)

Nit: "Attribute" instead of "Attirbute" -- not required to merge